### PR TITLE
POLIO-1501,  POLIO-1502,  POLIO-1503 campaign type filters

### PR DESCRIPTION
**Pull request by beygorghor**
Campaign type has been renamed to campaign categories, types is used for multi diseases campaigns now

Related JIRA tickets : POLIO-1501,  POLIO-1502,  POLIO-1503

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- renamed types filter to category filter
- hook to fetch types
- add new types filter

## How to test

Open campaigns or calendar page, and ply with types or categories filters

## Print screen / video

Screenshot 2024-03-13 at 15 54 31
Screenshot 2024-03-13 at 15 54 26
Screenshot 2024-03-13 at 15 54 18
Screenshot 2024-03-13 at 15 54 11

### CODE REVIEW

File: plugins/polio/js/src/domains/Calendar/campaignCalendar/Filters.tsx [hunk 1]
```diff
@@ -207,56 +204,72 @@ const Filters: FunctionComponent<Props> = ({
...
+                            value={campaignGroups}
+                            type="select"
+                            options={groupedCampaignsOptions}
+                            label={MESSAGES.groupedCampaigns}
+                        />
+                    )}
                 </Grid>
                 <Grid item xs={12} md={3}>
                     <InputComponent
                         loading={isFetchingCountries}
```
Comment (hakifran) [inline plugins/polio/js/src/domains/Calendar/campaignCalendar/Filters.tsx]: @beygorghor I don't understand why isFetchingCountries here ?

**Review by mathvdh — APPROVED** [review]
Ok for me but I'm really no frontend expert

File: plugins/polio/js/src/domains/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const pageWidth = 1980;
 
 const useStyles = makeStyles(theme => ({
     containerFullHeightNoTabPadded: {
-        ...commonStyles(theme).containerFullHeightNoTabPadded,
+        ...commonStyles(theme as Theme).containerFullHeightNoTabPadded,
```
Comment (hakifran) [inline plugins/polio/js/src/domains/Calendar/Calendar.tsx:50]: Why this theme as Theme?

**Review by hakifran — APPROVED** [review]
It looks good for me, apart from two comments(like questions) I made.

**Pull request merged**